include/hipSYCL/sycl/libkernel/builtins.hpp

```diff
@@ -967,6 +967,20 @@ HIPSYCL_DEFINE_BUILTIN(isnan, HIPSYCL_BUILTIN_OVERLOAD_SET_GENFLOATF,
 HIPSYCL_DEFINE_BUILTIN(isnan, HIPSYCL_BUILTIN_OVERLOAD_SET_GENFLOATD,
                        HIPSYCL_BUILTIN_GENERATOR_UNARY_T_RET_INT64)
 
+HIPSYCL_DEFINE_BUILTIN(isinf, HIPSYCL_BUILTIN_OVERLOAD_SET_GENFLOATF,
+                       HIPSYCL_BUILTIN_GENERATOR_UNARY_T_RET_INT)
+HIPSYCL_DEFINE_BUILTIN(isinf, HIPSYCL_BUILTIN_OVERLOAD_SET_GENFLOATD,
+                       HIPSYCL_BUILTIN_GENERATOR_UNARY_T_RET_INT64)
+
+HIPSYCL_DEFINE_BUILTIN(isfinite, HIPSYCL_BUILTIN_OVERLOAD_SET_GENFLOATF,
+                       HIPSYCL_BUILTIN_GENERATOR_UNARY_T_RET_INT)
+HIPSYCL_DEFINE_BUILTIN(isfinite, HIPSYCL_BUILTIN_OVERLOAD_SET_GENFLOATD,
+                       HIPSYCL_BUILTIN_GENERATOR_UNARY_T_RET_INT64)
+
+HIPSYCL_DEFINE_BUILTIN(signbit, HIPSYCL_BUILTIN_OVERLOAD_SET_GENFLOATF,
+                       HIPSYCL_BUILTIN_GENERATOR_UNARY_T_RET_INT)
+HIPSYCL_DEFINE_BUILTIN(signbit, HIPSYCL_BUILTIN_OVERLOAD_SET_GENFLOATD,
+                       HIPSYCL_BUILTIN_GENERATOR_UNARY_T_RET_INT64)
```

Comment:
Do those return correct types for vectors? E.g. `isfinite(vec<float,8>)` is supposed to return `vec<int32_t>` while `isfinite(vec<double,8>)` should return `vec<int64_t>`.

https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#table.relational.functions.vec

Reply:
They won't, I've just realised I fell on the trap of following the implementation of `isnan` a bit too religiously, so only scalars are supported...

Reply:
Excellent!